### ISSUE TITLE
New WKT rounding behaviour in 3.13

### ISSUE THREAD

**Issue by evetion** [libgeos/geos#1198]
Over at https://github.com/JuliaGeo/LibGEOS.jl, a Julia wrapper around GEOS, our WKT rounding tests failed on 3.13, probably due to https://github.com/libgeos/geos/pull/973. While I don't mind changes, I wondered if the following was intended:

Given
```julia
p = readgeom("POINT(0.12345 2.000)")
```

Print the point `p` for precisions from 0 to 10

Precision, output with `trim=true`
```
0, "POINT (0.1 2)"
1, "POINT (0.1 2)"
2, "POINT (0.1 2)"
3, "POINT (0.1 2)"
4, "POINT (0.1234 2)"
5, "POINT (0.12345 2)"
6, "POINT (0.12345 2)"
7, "POINT (0.12345 2)"
8, "POINT (0.12345 2)"
9, "POINT (0.12345 2)"
10, "POINT (0.12345 2)"
```

Precision, output with `trim=false`
```
0, "POINT (0 2)"
1, "POINT (0.1 2.0)"
2, "POINT (0.12 2.00)"
3, "POINT (0.123 2.000)"
4, "POINT (0.1235 2.0000)"
5, "POINT (0.12345 2.00000)"
6, "POINT (0.123450 2.000000)"
7, "POINT (0.1234500 2.0000000)"
8, "POINT (0.12345000 2.00000000)"
9, "POINT (0.123450000 2.000000000)"
10, "POINT (0.1234500000 2.0000000000)"
```

IMHO the `trim=true` behavior seems inconsistent and it might be good to describe the new behavior in the docs?

**Comment by mwtoews** [comment]
Thanks for detailed report, I'd agree with the observation with `trim=true` where the <3.13 output would have shown:
```
0, "POINT (0 2)"
1, "POINT (0.1 2)"
2, "POINT (0.12 2)"
3, "POINT (0.123 2)"
4, "POINT (0.1234 2)"
5, "POINT (0.12345 2)"
```
I'll take a look for a fix...

**Comment by mwtoews** [comment]
I've corrected my previous comment about expected output with precision=0, where non-zero `0.1` is shown for the first coordinate instead of `0`. This obviously dishonours zero precision, but the aim of the formatting is to show "pretty formatting" of all numbers. The formatter shows very small non-zero numbers in scientific precision, which is why a minimum of precision is kept. To honour precision=0, trim would need to be disabled.

Let me show a table of formatted numbers, where columns are the precision and row indexes are the values.

The first table is outputs as effectively formatted in GEOS 3.13, which shows the same issue described above:
```
               0       1        2         3          4
0.000012    1e-5  1.2e-5  1.23e-5  1.235e-5  1.2346e-5
0.000123  0.0001  0.0001   0.0001    0.0001     0.0001
0.001235   0.001   0.001    0.001     0.001     0.0012
0.012346    0.01    0.01     0.01      0.01     0.0123
0.123457     0.1     0.1      0.1       0.1     0.1235
1.234568       1     1.2     1.23     1.235     1.2346
```
And a third table is output with the pending fix in #1199:
```
               0       1        2         3          4
0.000012    1e-5  1.2e-5  1.23e-5  1.235e-5  1.2346e-5
0.000123  0.0001  0.0001   0.0001    0.0001     0.0001
0.001235   0.001   0.001    0.001     0.001     0.0012
0.012346    0.01    0.01     0.01     0.012     0.0123
0.123457     0.1     0.1     0.12     0.123     0.1235
1.234568       1     1.2     1.23     1.235     1.2346
```
Only three values in this table are impacted by the change. To see where they are, here are the non-zero differences:
```
                0        1        2        3        4
0.000012                                             
0.000123                                             
0.001235                                             
0.012346                               0.002         
0.123457                       0.02    0.023         
1.234568                                             
```

**Comment by evetion** [comment]
Thanks for the exhaustive explanation! It seems our test was quite an edge-case.

**Comment by mwtoews** [comment]
This fix will appear in the GEOS 3.13.1 patch release
